backend/qemu.pm

```diff
@@ -866,6 +866,9 @@ sub start_qemu ($self) {
         }
     }
 
+    my $v6checkhost = $vars->{IPV6CHECKHOST} // 'opensuse.org';
+    $vars->{IPV6AVAILABLE} = runcmd('ping', '-6', '-c', 1, '-w', 1, '-W', 1, $v6checkhost) ? 0 : 1;
```

Comment:
Doing this within the QEMU backend means the variable will only be available when using that specific backend. This is not ideal and also not in-line with adding the documentation under `.COMMON backend`.

I would also avoid running the command and setting the variable unconditionally before each test run as we don't always read the variable (and in fact don't read it at all within the scope of os-autoinst itself). So it would make more sense to add a test API function, e.g. `check_ipv6_available` that can be called as needed.

Maybe `check_ipv6_internet_connectivity` would also be a better name as this check will fail if IPv6 is available but but only for e.g. local connections.

Reply:
I adjusted the docs. Apparently I missed the correct table.

Regarding an test API change; I would like to avoid this. I understand that you mainly see variables as inputs for the backend but in this case I would argue that it should be fine to use them as output from the backend for the test (merely to provide capability information). What do others think?

Reply:
> Ah, so you're setting this variable to check its value later via `vars.json`

I intend to fetch this bit of information from within a running test with `get_var` - not sure if you consider this as »check its value later via `vars.json`« ;)

> But are you sure you only want this for QEMU-based tests?

Yes, for now. I looked into the baseclass (`backend::virt`) but it is used for a lot of other architectures where this check would not work or possible even produce wrong results (e.g. `pvm_hmc` and `spvm`). But you're right, it would also be feasible on other virtualized environments.